components/core/src/metrics.rs

```diff
@@ -71,9 +90,7 @@ fn statsd_client() -> Option<Client> {
 
 impl Counter {
     pub fn increment(&self) {
-        SEND_CHANNEL.lock()
-            .unwrap()
-            .clone()
+        get_sender()
```

Comment:
This is great - you might even be able to reduce the amount of clones per thread which would occur by calling this multiple times in a callstack by using [thread local sotrage](https://doc.rust-lang.org/std/thread/struct.LocalKey.html).